### PR TITLE
Enable uploading of media files to minio(s3 compatible)

### PR DESCRIPTION
**Pull request by urandu**
This is a PR to enable the upload of files to [minio](https://min.io/). Minio is s3 compatible and and be run locally using docker.

I have added the required settings on the settings in the UI and tested the functionality both locally and on a minio instance in the cloud.

This is related to [this issue](https://github.com/ccrisan/motioneye/issues/1485)

### CODE REVIEW

**Comment by ludiqfen** [comment]
Local

File: setup.py [hunk 1]
```diff
@@ -63,7 +63,7 @@ def apply_patches(self, base_dir):
 
     packages=['motioneye'],
 
-    install_requires=['tornado>=3.1,<6', 'jinja2', 'pillow', 'pycurl'],
+    install_requires=['tornado>=3.1,<6', 'jinja2', 'pillow', 'pycurl', 'minio'],
```
Comment (ccrisan) [inline setup.py]: Users shouldn't be forced to install `minio` in order to use motionEye. Having it as an option is fine, but let's not add it as a dependency.

File: setup.py [hunk 2]
```diff
@@ -63,7 +63,7 @@ def apply_patches(self, base_dir):
 
     packages=['motioneye'],
 
-    install_requires=['tornado>=3.1,<6', 'jinja2', 'pillow', 'pycurl', 'minio'],
```
Comment (ccrisan) [inline setup.py]: The fact that you removed `minio` from dependencies is not enough for this to work. You still import it in the python code. You should import it in a `try` .. `except` clause.

**Comment by MichaIng** [comment]
@urandu 
Many thanks for your PR. Could you review [this commit](https://github.com/motioneye-project/motioneye/commit/bc54c5d#diff-19976dd03886cacaf68b9d122696faf2ffc677d41b96b0d8eaca47858093c8b3) whether the added S3 upload service is compatible with Minio, or otherwise can be made compatible to support both, AWS and Minio?

**Comment by MichaIng** [comment]
Marking this as closed in favour of: #2376
However, many thanks for this PR, which helped me to get the generic S3 compatible upload service up.

**Comment by MichaIng** [comment]
Marking this as closed in favour of: #2376
However, many thanks for this PR, which helped me to get the generic S3 compatible upload service up.

**Pull request closed without merging**